### PR TITLE
Offline storage

### PR DESCRIPTION
**Pull request by Oxydeas**
This adds multiple helper functions in the LocalSotrage class to store and retrieve Events, ShopItems and Tasks. Fully tested except Tasks as they do not have the correct structure yet.

### CODE REVIEW

**Review by sansive — APPROVED** [review]
Nice work!

File: app/src/main/java/com/github/houseorganizer/houseorganizer/storage/LocalStorage.java [hunk 1]
```diff
@@ -1,21 +1,204 @@
...
+            InputStreamReader isr = new InputStreamReader(fis);
+            BufferedReader bufferedReader = new BufferedReader(isr);
+            StringBuilder buffer = new StringBuilder();
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                buffer.append(line);
+            }
+            return buffer.toString();
+        } catch (Exception e) {
+            return "";
```
[the remark this answers is not in the document]
Comment (Arkadyus) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/storage/LocalStorage.java:88]: I think it sounds right to return nothing if we cant read the cache file, it just means we did not get to recover any data

**Review by Arkadyus — APPROVED** [review]
Everything looks good, nice work

File: app/src/main/java/com/github/houseorganizer/houseorganizer/storage/LocalStorage.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/storage/LocalStorage.java:88]: Yes, my reasoning was that of Thibaud

**Pull request merged**